cloud/shared/bin/doctor_impl

```diff
@@ -0,0 +1,221 @@
+            https://docs.oracle.com/javase/9/install/installation-jdk-and-jre-linux-platforms.htm#JSJIG-GUID-737A84E4-2EFF-4D38-8E60-3E29D1B884B8
+        """
+
+    def remedy_mac(self):
+        return """
+            Keytool is usually installed with Java. If this check is failing but the Java JRE check is not, you may need to try reinstalling Java with a newer version.
+            https://docs.oracle.com/javase/9/install/installation-jdk-and-jre-macos.htm#JSJIG-GUID-2FE451B0-9572-4E38-A1A5-568B77B146DE
+        """
+
+class AwsCliCheck(Check):
+    remedy = """
+        Need to install the AWS CLI tool.
+        https://docs.aws.amazon.com/cli/latest/userguide/getting-started-install.html
+    """
+
+    def name(self):
+        return "AWS CLI"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() not in ['aws', 'azure']
+
+    def is_ok(self):
+        return self._check_command_succeeds(["aws", "help"])
+
+class AzureCliCheck(Check):
+    remedy = """
+        Need to install the Azure CLI tool.
+        https://docs.microsoft.com/en-us/cli/azure/install-azure-cli
+    """
+
+    def name(self):
+        return "Azure CLI"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() != "azure"
+
+    def is_ok(self):
+        return self._check_command_succeeds(["az"])
+
+class AwsLoginCheck(Check):
+    remedy = """
+        Need to configure AWS CLI tool. Run `aws configure` and enter credentials
+        or for other options visit
+        https://docs.aws.amazon.com/cli/latest/userguide/cli-configure-quickstart.html
+    """
+
+    def name(self):
+        return "AWS login"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() not in ['aws', 'azure']
+
+    def is_ok(self):
+        output = self._get_command_output(["aws", "configure", "list-profiles"])
+
+        # the default profile will be present if they are logged in
+        return output and 'default' in output.stdout.decode('ascii')
+
+class AzureLoginCheck(Check):
+    remedy = """
+        Need to authenticate with the Azure CLI tool.
+        https://docs.microsoft.com/en-us/cli/azure/authenticate-azure-cli
+    """
+
+    def name(self):
+        return "Azure login"
+
+    def should_skip(self):
+        return os.environ.get('CIVIFORM_CLOUD_PROVIDER', "").lower() != "azure"
+
+    def is_ok(self):
+        return self._check_command_succeeds(["az", "ad", "signed-in-user", "show"])
+
+checks = [
+    DockerCheck(),
+    TerraformCheck(),
+    JavaCheck(),
+    KeytoolCheck(),
+    AwsCliCheck(),
+    AzureCliCheck(),
+    AwsLoginCheck(),
+    AzureLoginCheck()
+]
+
+problems = []
+for check in checks:
+    if check.should_skip():
+        continue
+
+    print(check.name().ljust(30, "."), end="", flush=True)
+
+    if check.is_ok():
+        print("OK")
+        continue
```

Comment:
why use continue here instead of an else?

Reply:
Mostly aesthetic I guess, looks nicer to my eyes with less indentation and syntax